File: SDKs/Aspose.Pdf_Cloud_SDK_for_.NET/src/Com/Aspose/PDF/Model/Signature.cs
using System;
using System.Text;
using System.Collections;
using System.Collections.Generic;

namespace Com.Aspose.PDF.Model {
  public class Signature {
    public string SignaturePath { get; set; }

    public string SignatureType { get; set; }

    public string Password { get; set; }

    public string Contact { get; set; }

    public string Location { get; set; }

    public bool? Visible { get; set; }

    public Rectangle Rectangle { get; set; }

    public string FormFieldName { get; set; }

    public string Authority { get; set; }

    public string Date { get; set; }

    public override string ToString()  {
      var sb = new StringBuilder();
      sb.Append("class Signature {\n");
      sb.Append("  SignaturePath: ").Append(SignaturePath).Append("\n");
      sb.Append("  SignatureType: ").Append(SignatureType).Append("\n");
      sb.Append("  Password: ").Append(Password).Append("\n");
      sb.Append("  Contact: ").Append(Contact).Append("\n");
      sb.Append("  Location: ").Append(Location).Append("\n");
      sb.Append("  Visible: ").Append(Visible).Append("\n");
      sb.Append("  Rectangle: ").Append(Rectangle).Append("\n");
      sb.Append("  FormFieldName: ").Append(FormFieldName).Append("\n");
      sb.Append("  Authority: ").Append(Authority).Append("\n");
      sb.Append("  Date: ").Append(Date).Append("\n");
      sb.Append("}\n");
      return sb.ToString();
    }
  }
  }

File: SDKs/Aspose.Pdf-Cloud-SDK-for-Ruby/lib/aspose_pdf_cloud/models/signature.rb
module AsposePDFCloud
  # 
  class Signature < BaseObject
    attr_accessor :signature_path, :signature_type, :password, :contact, :location, :visible, :rectangle, :form_field_name, :authority, :date
    # attribute mapping from ruby-style variable name to JSON key
    def self.attribute_map
      {
        
        # The SignaturePath
        :'signature_path' => :'SignaturePath',
        
        # The SignatureType
        :'signature_type' => :'SignatureType',
        
        # The Password
        :'password' => :'Password',
        
        # The Contact
        :'contact' => :'Contact',
        
        # The Location
        :'location' => :'Location',
        
        # The Visible
        :'visible' => :'Visible',
        
        # The Rectangle
        :'rectangle' => :'Rectangle',
        
        # The FormFieldName
        :'form_field_name' => :'FormFieldName',
        
        # The Authority
        :'authority' => :'Authority',
        
        # The Date
        :'date' => :'Date'
        
      }
    end

    # attribute type
    def self.swagger_types
      {
        :'signature_path' => :'String',
        :'signature_type' => :'String',
        :'password' => :'String',
        :'contact' => :'String',
        :'location' => :'String',
        :'visible' => :'BOOLEAN',
        :'rectangle' => :'Rectangle',
        :'form_field_name' => :'String',
        :'authority' => :'String',
        :'date' => :'String'
        
      }
    end

    def initialize(attributes = {})
      return if !attributes.is_a?(Hash) || attributes.empty?

      # convert string to symbol for hash key
      attributes = attributes.inject({}){|memo,(k,v)| memo[k.to_sym] = v; memo}

      
      if attributes[:'SignaturePath']
        self.signature_path = attributes[:'SignaturePath']
      end
      
      if attributes[:'SignatureType']
        self.signature_type = attributes[:'SignatureType']
      end
      
      if attributes[:'Password']
        self.password = attributes[:'Password']
      end
      
      if attributes[:'Contact']
        self.contact = attributes[:'Contact']
      end
      
      if attributes[:'Location']
        self.location = attributes[:'Location']
      end
      
      if attributes[:'Visible']
        self.visible = attributes[:'Visible']
      end
      
      if attributes[:'Rectangle']
        self.rectangle = attributes[:'Rectangle']
      end
      
      if attributes[:'FormFieldName']
        self.form_field_name = attributes[:'FormFieldName']
      end
      
      if attributes[:'Authority']
        self.authority = attributes[:'Authority']
      end
      
      if attributes[:'Date']
        self.date = attributes[:'Date']
      end
      
    end

  end
end
